src/location.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useEffect, useState } from "react";
+import { StyleSheet, Text, View, Button } from "react-native";
+import * as TaskManager from "expo-task-manager";
+import * as Location from "expo-location";
+
+const LOCATION_TASK_NAME = "LOCATION_TASK_NAME";
+let foregroundSubscription: any = null;
+
+// Define the background task for location tracking
+TaskManager.defineTask(
+  LOCATION_TASK_NAME,
+  async ({ data, error }: { data: any; error: any }) => {
+    if (error) {
+      console.error(error);
+      return;
+    }
+    if (data) {
+      // Extract location coordinates from data
+      const { locations } = data;
+      const location = locations[0];
+      if (location) {
+        console.log(new Date(), "Location in background", location.coords);
+      }
+    }
+  }
+);
+
+export default function LTest() {
```

Comment:
Some comments on this function would help :)

Reply:
WIP, but will add comments for location stuff in next PR